bindings/pyroot/ROOT.py

```diff
@@ -274,29 +274,59 @@ def _TTreeAsMatrix(self, columns=None, dtype="double"):
     # Get all columns of the tree if no columns are specified
     if columns is None:
         columns = [branch.GetName() for branch in self.GetListOfBranches()]
+    if not isinstance(columns, list):
```

Comment:
Do we really need this check or can we just rely on columns being an iterable?

Reply:
Do you mean by "rely on" that we just skip the error-handling? Actually checking whether an object is iterable not straight forward. Type-checking in python is a never-ending story...

Reply:
If we take the pandas dataframe as example, in the documentation of [as_matrix](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.as_matrix.html) they specify that the method receives a list, but if you pass e.g. a tuple it works as well. If you pass something that is not iterable, it just crashes when the code tries to iterate. I would follow the same strategy.

Reply:
Some kind of "implicit" error-handling. Fine for me!